install.sh

```diff
@@ -262,6 +262,59 @@ outdated_glibc() {
   version_lt "$glibc_version" "$REQUIRED_GLIBC_VERSION"
 }
 
+usage() {
+  cat <<EOS
+Homebrew Installer
+Usage: /bin/bash install.sh [--help] [--brew-git-remote URL] [--core-git-remote URL]
+
+This script installs Homebrew, the missing package manager for macOS (or Linux).
+
+Options:
+  -b, --brew-git-remote=URL   Use this URL as the Homebrew/brew git remote.
+                              Default: ${HOMEBREW_BREW_DEFAULT_GIT_REMOTE}.
+
+  -c, --core-git-remote=URL   Use this URL as the Homebrew/homebrew-core git remote.
+                              Default: ${HOMEBREW_CORE_DEFAULT_GIT_REMOTE}.
+
+  -h, --help                  Show this help message and exit.
+EOS
+  exit "${1:-0}"
+}
+
+# Parse remote URLs of Homebrew repositories from command line arguments if presented.
+unset HOMEBREW_{BREW,CORE}_GIT_REMOTE  # unset variables from environment at first
+# The value of the script basename in different situations:
+#   - "bash":       invoked by `bash -c` with no arguments.
+#   - "install.sh": invoked by bash directly.
+#   - otherwise:    invoked by `bash -c` with at least one argument.
+if [[ "$(basename -- "$0")" != "bash" && "$(basename -- "$0")" != "install.sh" ]]; then
```

Comment:
`bash -c 'basename -- "$0"'` returns `bash`, but `sh -c 'basename -- "$0"'` returns `sh` and `zsh` for `zsh`, etc. I haven't found an approach to write universal codes for all POSIX-compliant shells. I assume users always use `bash`.

Reply:
The shebang is `#!/bin/bash`, plus the code snippet for installing `brew` tells users to run the script using `/bin/bash`, so I think that's a safe assumption.

Reply:
`install.sh` requires specifically BASH and disallows other shells.

Reply:
I find a way to determine which shell we are using with `ps -p $$ -ucomm=''` (tested on macOS Catalina and Ubuntu 16.04/20.04). And I add an assertion for BASH here:
```bash
SHELL_NAME="$(ps -p $$ -oucomm='' | head -n 1 | tr -d ' ')"
if [[ -z "${BASH_VERSION:-}" || "${SHELL_NAME}" != "bash" ]]; then
  warn "BASH is required to interpret this script."; usage 1
fi
```
Without this assertion, invoke `install.sh` with `zsh` or `sh` do not get any error, but stuck at waiting for user input here:
> Press RETURN to continue or any other key to abort

(`sh` is symlinked to `bash` on macOS, but on Ubuntu they are different shell.)

Reply:
I think `-n "${BASH_VERSION:-}"` is enough. We don't need the shell name at all.